templates/problem/list.html

```diff
@@ -287,6 +290,15 @@ <h3>{{ _('Hot problems') }} <i class="fa fa-fire"></i></h3>
                         {% if not request.in_contest %}
                             <td class="ac-rate">{{ problem.ac_rate|floatformat(1) }}%</td>
                         {% endif %}
+                        {% if problem.has_public_editorial() %}
```

Comment:
This will do O(pagination size) number of queries. See the above comment for a possible solution. Also, what happens when the user is in contest?